app/src/main/java/uk/ac/bath/petmatch/MainActivity.java

```diff
@@ -116,13 +118,20 @@ protected void generateLoggedUserView() {
                 loggedUserEmail.setText(loginService.getLoggedInUser().getEmail());
                 Menu nav_Menu = navigationView.getMenu();
                 nav_Menu.findItem(R.id.nav_user_capabilities).setVisible(true);
+
+                if(loginService.getLoggedInUser().getShelter() != null) {
```

Comment:
getLoggedInUser() could be null, so get shelter() on it could result in null pointer exception